version: 1.0

description: >
  A workflow example that illustrates error handling. By default when any task fails,
  the notify_on_error task will be executed and the workflow will transition to the
  failed state.

input:
  - cmd

output:
  stdout: <% $.stdout %>

tasks:
  task1:
    action: core.local cmd=<% $.cmd %>
    on-complete:
      - if: <% state() = "succeeded" %>
        publish: stdout=<% result().stdout %>
      - if: <% state() = "failed" %>
        publish: stderr=<% result().stderr %>
        next: notify_on_error
  notify_on_error:
    action: core.local cmd="printf '<% $.stderr %>'"
    on-complete:
      - next: fail
